InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -110,6 +110,17 @@ QList<Optional<TupleSet> > CompositeQuery::execute(QList<TupleSet> input)
 	return outNode_->calculatedOutputs_;
 }
 
+Query* CompositeQuery::addQuery(std::unique_ptr<Query> query)
+{
+	auto rawQ = query.get();
+	// A query should only be added once:
+	Q_ASSERT(std::find_if(nodes_.begin(), nodes_.end(), [rawQ](QueryNode* existing) {return existing->q_.get() == rawQ;})
+					== nodes_.end());
+	auto newNode = new QueryNode(std::move(query));
```

Comment:
So I read that in detail and thought about it, and I still think `&&` is the right way:
- There are a few subtle differences that you can read in the comments that point in `&&`'s favor. One of them is the performance penalty you pay for an extra temporary creation which you don't need.
- The author's only argument for using values is that ownership transfer is guaranteed. But why is that important? In general, every time you see a `std::move` you must assume that whatever you're moving is now gone. Whether that's actually so or not, is irrelevant and you certainly shouldn't go ahead and determine whether the value you're left with is still the old one.